iothub/device/src/IotHubBaseClient.cs

```diff
@@ -131,10 +131,10 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         }
 
         /// <summary>
-        /// Sets a new delegate for the connection status changed callback. If a delegate is already associated,
-        /// it will be replaced with the new delegate.
+        /// Sets a new listener for the connection status change callback. If a listener is already associated,
+        /// it will be replaced with the new listener.
         /// </summary>
-        /// <param name="statusChangeHandler">The name of the method to associate with the delegate.</param>
+        /// <param name="statusChangeHandler">The listener for the connection status change callback.</param>
```

Comment:
We should pick what term we like most and stick with it. Listener? Callback? Handler?

Across libraries, APIs, and comments/variable names.

Reply:
That's a good idea. I like the term "listener". We had used it in our pnp-fied API comments, and it had been received well. So, I had been updating our APIs with that as I brought over the doc comments.

Reply:
I'm not sure if it is very C#-ish though.

Reply:
Inquire with the Azure SDK team?

Reply:
This is the recommendation I received: https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#names-of-common-types
Based on this, callback would be the recommended suffix and term for our scenario.

Reply:
https://github.com/Azure/azure-iot-sdk-csharp/pull/2760